api/utils_celery.py

```diff
@@ -59,9 +58,15 @@ def get_inactive_task_pages():
 
 
 def process_changed_articles():
-    for language, page_title in iter_changed_pages(logger):
+    for language, page_title, change in iter_changed_pages(logger):
         # print(len(get_inactive_task_pages()))
-        if page_title not in get_inactive_task_pages():
-            streamer.info(f"EVENT: {page_title} ({language})")
+
+        # Delete pickle file if the page is deleted.
+        if change.get('type') == 'log' and change.get('log_type') == 'delete' and change.get('log_action') == 'delete':
+            streamer.info(f"EVENT (delete): {page_title} ({language})")
+            process_article_deletion.delay(language, page_title, change.get('log_id'))
+        # Otherwise, queue the article for processing
+        elif page_title not in get_inactive_task_pages():
+            streamer.info(f"EVENT (edit): {page_title} ({language})")
```

Comment:
can we do better with the logging than hard-coding `edit` and `delete` as the only types? i assumed when looking at the log stream that it was pulling the event type from the stream data, but from here it looks like it's just doing handling delete explicitly and assuming every other event is an edit, but I don't know whether there are other explicit (perhaps unhandled) event types coming through the stream.

Reply:
Sure! We can pass in the event type in this message. There is another type called "new" that we process when new articles are created, but how we react is the same as "edit" (it's still an edit, after all!) so I figured for logging purposes they could be bundled.

Reply:
Done. Merging! I'll keep an eye on the logs to ensure everything is working properly.